### PR TITLE
Enhanced logs for advanced auth capabilities

### PR DESCRIPTION
**Pull request by idanovo**

### CODE REVIEW

File: pkg/httputil/tracing.go [hunk 1]
```diff
@@ -134,19 +138,25 @@ func TracingMiddleware(requestIDHeaderName string, fields logging.Fields, traceR
...
 
 			traceFields := logging.Fields{
-				"took":             time.Since(startTime),
-				"status_code":      responseWriter.StatusCode,
-				"sent_bytes":       responseWriter.ResponseSize,
-				"request_body":     presentBody(requestBodyTracer.bodyRecorder.Buffer),
-				"response_body":    presentBody(responseWriter.BodyRecorder.Buffer),
-				"response_headers": responseWriter.Header(),
+				"start_time":  startTime,
+				"end_time":    time.Now(),
```
Comment (Isan-Rivkin) [inline pkg/httputil/tracing.go:142]: Why added that new field start_time? (and end_time)

File: pkg/config/config.go [hunk 2]
```diff
@@ -582,6 +587,10 @@ func (c *Config) IsExternalPrincipalsEnabled() bool {
 	return c.IsAuthTypeAPI() && c.Auth.AuthenticationAPI.ExternalPrincipalsEnabled
 }
 
+func (c *Config) IsAdvancedAuth() bool {
+	return c.IsAuthTypeAPI() && c.IsAuthExternalOrInternal()
```
Comment (Isan-Rivkin) [inline pkg/config/config.go:591]:
```suggestion
	return c.IsAuthTypeAPI() && ( c.Auth.UIConfig.RBAC == AuthRBACExternal || c.Auth.UIConfig.RBAC == AuthRBACInternal )
```

File: pkg/config/config.go [hunk 3]
```diff
@@ -570,9 +570,14 @@ func (c *Config) IsAuthUISimplified() bool {
 	return c.Auth.UIConfig.RBAC == AuthRBACSimplified
 }
 
+func (c *Config) IsAuthExternalOrInternal() bool {
+	return c.Auth.UIConfig.RBAC == AuthRBACExternal || c.Auth.UIConfig.RBAC == AuthRBACInternal
+}
```
Comment (Isan-Rivkin) [inline pkg/config/config.go:575]:
```suggestion

```

File: pkg/api/auth_middleware.go [hunk 4]
```diff
@@ -105,7 +105,10 @@ func AuthMiddleware(logger logging.Logger, swagger *openapi3.Swagger, authentica
 				return
 			}
 			if user != nil {
-				ctx := logging.AddFields(r.Context(), logging.Fields{logging.UserFieldKey: user.Username})
+				ctx := r.Context()
+				if isAdvancedAuth {
+					ctx = logging.AddFields(ctx, logging.Fields{logging.UserFieldKey: user.Username})
```
Comment (Isan-Rivkin) [inline pkg/api/auth_middleware.go:110]: IIUC This should stay X-LakeFS-Client is the enhanced capability that can be customized with data (`client` field in audit log) 
While this field `user` is just the authenticated lakeFS user.

**Review by Isan-Rivkin — CHANGES_REQUESTED** [review]

**Review by Isan-Rivkin — CHANGES_REQUESTED** [review]
Hey, you asked for re-review but didnt actually commit or address everything please see again :)

**Comment by idanovo** [comment]
> Hey, you asked for re-review but didnt actually commit or address everything please see again :)
@Isan-Rivkin I did fix everything except for your comment about the `user` field which is still waiting for a response from product

**Comment by Isan-Rivkin** [comment]
> > Hey, you asked for re-review but didnt actually commit or address everything please see again :)
> 
> @Isan-Rivkin I did fix everything except for your comment about the `user` field which is still waiting for a response from product

Are you [sure](https://github.com/treeverse/lakeFS/pull/8051/files#r1704185126)? :)

File: pkg/httputil/tracing.go (hunk 1 above)
Comment (idanovo) [reply pkg/httputil/tracing.go:142]: damm forgot to commit this change

**Review by Isan-Rivkin — APPROVED** [review]
LGTM

**Pull request merged**